### PR TITLE
Admin users

### PR DESCRIPTION
**Pull request by sCar3Cr0w25**
Hau san this is the pull request to get your comments on what I need to change.

### CODE REVIEW

File: app/assets/images/default_user.jpg:Zone.Identifier [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ZoneTransfer]
```
Comment (CW-HauNC) [inline app/assets/images/default_user.jpg:Zone.Identifier:1]: remove this file

File: app/controllers/admin/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    def destroy
+      @post.destroy
+
+      redirect_to admin_posts_path
+    end
+
+    def search
+      @results = @q.result(distinct: true)
+    end
```
Comment (CW-HauNC) [inline app/controllers/admin/posts_controller.rb:46]: remove method search, handle list and search in index

File: app/models/post.rb [hunk 3]
```diff
@@ -1,4 +1,13 @@
+# frozen_string_literal: true
+
 class Post < ApplicationRecord
-    has_many: comments
-    belongs_to: user, group
+
+  paginates_per 1
```
Comment (CW-HauNC) [inline app/models/post.rb:5]: change per to 10

File: app/models/user.rb [hunk 4]
```diff
@@ -1,7 +1,40 @@
+# frozen_string_literal: true
+
 class User < ApplicationRecord
   extend Devise::Models
+
+  paginates_per 2
```
Comment (CW-HauNC) [inline app/models/user.rb:6]: change per to 10

File: app/models/user.rb [hunk 5]
```diff
@@ -1,7 +1,40 @@
...
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
+
+  has_one_attached :avatar
+
+  validates :avatar, file_size: { less_than_or_equal_to: 5.megabytes },
+                     file_content_type: { allow: ['image/jpeg', 'image/png', 'image/gif'] }
+  validate :birthday_cannot_be_in_the_future
+  def birthday_cannot_be_in_the_future
+    errors.add(:birthday, "can't be in the past") if birthday.present? && birthday > Date.today
+  end
```
Comment (CW-HauNC) [inline app/models/user.rb:29]: change this method to private method. Because only use in this model

File: app/views/admin/posts/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+       <% @posts.each do |post| %>
+         <tr>
+           <td><%= post.title %></td>
+           <td><%= post.user.username %></td>
+           <td><%= post.content %></td>
+           <td><%= post.created_at %></td>
+           <td><%= link_to 'Show', admin_post_path(post) %></td>
+           <td><%= link_to 'Edit', edit_admin_post_path(post) %></td>
+           <td><%= link_to "Destroy", admin_post_path(post), data: { turbo_method: :delete, turbo_confirm: "Are you sure?" } %></td>
+         </tr>
```
Comment (CW-HauNC) [inline app/views/admin/posts/index.html.erb:22]: using render partial

File: app/views/admin/posts/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+       <% end %>
+      </tbody>
+   </table>
+   
+   <%= paginate @posts %>
```
Comment (CW-HauNC) [inline app/views/admin/posts/index.html.erb:27]: missing new line at the end of file

File: app/views/admin/posts/search.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<table class="table table-bordered border-primary">
```
Comment (CW-HauNC) [inline app/views/admin/posts/search.html.erb:1]: Remove this file => just use index

File: app/views/admin/users/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  <% if @user.avatar.present? %>
+    <%= image_tag @user.avatar, width: 100 %>
+  <% end %>
+  </div>
+
+  <br>
+
+    <%= f.submit "Create user", class: "btn btn-outline-primary" %>
+    <% end %>
+</div>
```
Comment (CW-HauNC) [inline app/views/admin/users/_form.html.erb:53]: missing new line at the end of file

File: app/views/admin/users/index.html.erb [hunk 10]
```diff
@@ -1,2 +1,50 @@
...
+          <% end %>
+        </td>
+           <td><%= user.username %></td>
+           <td><%= user.email %></td>
+           <td><%= user.birthday %></td>
+           <td><%= user.status %></td>
+           <td><%= link_to 'Show', admin_user_path(user) %></td>
+           <td><%= link_to 'Edit', edit_admin_user_path(user) %></td>
+           <td><%= link_to "Destroy", admin_user_path(user), data: { turbo_method: :delete, turbo_confirm: "Are you sure?" } %></td>
+         </tr>
```
Comment (CW-HauNC) [inline app/views/admin/users/index.html.erb:32]: using render partial

File: app/controllers/admin/base_controller.rb [hunk 11]
```diff
@@ -1,11 +1,17 @@
...
     before_action :check_admin_permission
 
     private
 
     def check_admin_permission
-      redirect_to homepage_path unless current_user.admin?
+      redirect_to root_path unless current_user.admin?
+    end
+
+    def get_user
```
Comment (CW-AnhNH) [inline app/controllers/admin/base_controller.rb:13]: move this method to admin/users_controller

File: app/controllers/admin/posts_controller.rb [hunk 12]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Admin
+  class PostsController < Admin::BaseController
+    before_action :get_post, only: %i[show edit update destroy]
+    before_action :set_q, only: %i[index]
+
+    def index
+      # @posts = Post.page(params[:page]).all
```
Comment (CW-AnhNH) [inline app/controllers/admin/posts_controller.rb:9]: remove

File: app/controllers/admin/users_controller.rb [hunk 13]
```diff
@@ -1,5 +1,54 @@
+# frozen_string_literal: true
+
 module Admin
   class UsersController < Admin::BaseController
-    def index; end
+    before_action :get_user, only: %i[show edit update destroy]
+    before_action :set_q, only: %i[index]
+    def index
+      # @users = User.page(params[:page]).where.not(admin: true)
```
Comment (CW-AnhNH) [inline app/controllers/admin/users_controller.rb:8]: remove unused code

File: app/controllers/admin/users_controller.rb [hunk 14]
```diff
@@ -1,5 +1,54 @@
+# frozen_string_literal: true
+
 module Admin
   class UsersController < Admin::BaseController
-    def index; end
+    before_action :get_user, only: %i[show edit update destroy]
+    before_action :set_q, only: %i[index]
```
Comment (CW-AnhNH) [inline app/controllers/admin/users_controller.rb:6]: add empty line
```ruby
before_action :set_q, only: %i[index]

def index

```

File: app/views/admin/posts/_search.html.erb [hunk 15]
```diff
@@ -0,0 +1,8 @@
+<%= search_form_for @q, url: admin_posts_path do |f| %>
+    <%= f.search_field :title_or_content_cont %>
+    <%= f.label :Public %>
+    <%= f.check_box :privacy_eq_any, { multiple: true }, 0, '' %>
+    <%= f.label :Private %>
+    <%= f.check_box :privacy_eq_any, { multiple: true }, 1, '' %>
+    <%= f.submit "Search", class: "btn btn-outline-info" %>
+<% end %>
```
Comment (CW-AnhNH) [inline app/views/admin/posts/_search.html.erb:8]: add newline at the end

File: app/views/admin/users/edit.html.erb [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+</div>
+<div>
+<%= image_tag("preview.png", :id => "image_preview", width: 80) %>
+</div>
+<br>
+  <%= f.submit "Update", class: "btn btn-outline-primary" %>
+  <% end %>
+  <%= link_to 'Back', admin_posts_path %>
+
+
```
Comment (CW-AnhNH) [inline app/views/admin/users/edit.html.erb:47]: remove

**Pull request merged**